### PR TITLE
MAISTRA-1634: Add gen-check presubmit test for maistra/istio

### PR DESCRIPTION
**Pull request by jwendell**
Builder image need new tools because of that

### CODE REVIEW

File: prow/config.gen.yaml [hunk 1]
```diff
@@ -447,6 +447,32 @@ presubmits:
...
+    branches:
+      - maistra-2.0
+    rerun_command: /test gencheck
+    spec:
+      containers:
+      - image: "quay.io/maistra-dev/maistra-builder:2.0"
+        imagePullPolicy: Always
+        command:
+        - make
+        - gen-check
```
Comment (bison) [inline prow/config.gen.yaml:465]: This is great. I actually recently added targets to run the code-gen on our additional `ServiceMesh` bits. I wanted to keep the changes to the upstream `Makefile` minimal though, so it's not run as part of the normal `make gen` target. Can we maybe have this bit read:

```yaml
- make
- maistra-gen-k8s-client
- gen-check
```

That should catch our stuff as well.

Comment (jwendell) [reply prow/config.gen.yaml:465]: Will this target just generate files or it's a check that will fail if the PR author didn't generate them? Note that upstream has `gen` and `gen-check`. The latter is meant to fail and to be used in prow.

Comment (bison) [reply prow/config.gen.yaml:465]: This target just generates the files, but the `gen-check` target works by first running `gen` then running a script that checks that the git working directory is clean. So this should be enough.

Comment (jwendell) [reply prow/config.gen.yaml:465]: gotcha, ok, will add that

**Pull request merged**